src/ucs/config/parser.c

```diff
@@ -1374,6 +1449,11 @@ ucs_status_t ucs_config_parser_fill_opts(void *opts, ucs_config_field_t *fields,
         goto err_free;
     }
 
+    status = ucs_config_parse_config_file(UCX_CONF_FILE, 0);
+    if (status != UCS_OK) {
+        goto err_free;
+    }
```

Comment:
What do you mean? `ucs_config_apply_config_vars` is called only from here, and from a specific gtest which overrides it in order to test.

Reply:
i think we should call ucs_config_parse_config_file (in the init_once) **before** calling ucs_config_apply_config_vars.
because seems that now it would work only for gtest, but not for "real" config file

Reply:
Why? It does work for the "real" config file, verified that.

Reply:
AFAIU, ucs_config_parse_config_file populates tha hash, and ucs_config_apply_config_vars uses that hash.
so if ucs_config_apply_config_vars is called before ucs_config_parse_config_file , the hash would still be empty?

Reply:
Moved it